### PR TITLE
feat: add keyboard navigation for multiselect

### PR DESCRIPTION
**Pull request by abhishekashyap**
### PR Checklist

Please check if your PR fulfills the following requirements:

-   [x] Tests for the changes have been added (for bug fixes/features)
-   [ ] Docs have been added/updated (for bug fixes/features)

### PR Type

What kind of change does this PR introduce?

-   [ ] Bugfix
-   [x] Feature
-   [ ] Code style update (formatting, local variables)
-   [ ] Refactoring (no functional changes, no api changes)
-   [ ] Build related changes
-   [ ] CI related changes
-   [ ] Performance improves
-   [ ] Adding missing tests
-   [ ] Documentation content changes
-   [ ] Other... Please describe:

### Does this PR introduce a breaking change?

-   [ ] Yes
-   [x] No

### Additional context

Provide any additional context or screenshots about the feature PR.

### CODE REVIEW

File: packages/core/src/components/CheckboxGroup/CheckboxGroup.tsx [hunk 1]
```diff
@@ -74,6 +79,14 @@ const Component: FC<CheckboxGroupProps> = memo(
                 },
                 [values, onChange]
             ),
+            handleSelectionFromKeyboard = useCallback(
```
Comment (gmukul01) [inline packages/core/src/components/CheckboxGroup/CheckboxGroup.tsx:82]: Can we extract the common logic from `handleOptionClick` & `handleSelectionFromKeyboard` ?

File: packages/core/src/components/Modal/Modal.tsx [hunk 2]
```diff
@@ -50,10 +51,16 @@ const Component: FC<ModalProps> = memo(
             open && isEscPressed && onCloseModal && onCloseModal();
         }, [open, isEscPressed]);
 
+        const tempHandler = () => {
```
Comment (gmukul01) [inline packages/core/src/components/Modal/Modal.tsx:54]: Could we please remove all the changes in this file ?

File: packages/core/src/components/MultiSelect/Options/Options.tsx [hunk 3]
```diff
@@ -26,6 +37,13 @@ const Component: FC<OptionsProps> = memo(props => {
             },
             [selectedValues, onOptionClick]
         ),
+        handleSelectionFromKeyboard = useCallback(
```
Comment (gmukul01) [inline packages/core/src/components/MultiSelect/Options/Options.tsx:40]: Can we extract out the common logic from `handleCheckboxClick` & `handleSelectionFromKeyboard`.

**Review by gmukul01 — CHANGES_REQUESTED** [review]

File: packages/core/src/components/Modal/Modal.tsx (hunk 2 above)
Comment (abhishekashyap) [reply packages/core/src/components/Modal/Modal.tsx]: Reverted

**Review by gmukul01 — APPROVED** [review]
LGTM

**Pull request merged**